metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -0,0 +1,1152 @@
+      case ServerCommands.AnalyzeStacktrace(content) =>
+        Future {
+          // Getting the service for focused document and first one otherwise
+          val service =
+            focusedDocument.map(getServiceFor).getOrElse(folderServices.head)
+          val command = service.analyzeStackTrace(content)
+          command.foreach(languageClient.metalsExecuteClientCommand)
+          scribe.debug(s"Executing AnalyzeStacktrace ${command}")
+        }.asJavaObject
+
+      case ServerCommands.GotoSuperMethod(textDocumentPositionParams) =>
+        getServiceFor(textDocumentPositionParams.getTextDocument().getUri())
+          .gotoSupermethod(textDocumentPositionParams)
+
+      case ServerCommands.SuperMethodHierarchy(textDocumentPositionParams) =>
+        scribe.debug(s"Executing SuperMethodHierarchy ${params.getCommand()}")
+        getServiceFor(textDocumentPositionParams.getTextDocument().getUri())
+          .superMethodHierarchy(textDocumentPositionParams)
+
+      case ServerCommands.ResetChoicePopup() =>
+        def resetPopupChoice(value: String) =
+          collectSeq(_.resetPopupChoice(value))(_ => ())
+        val argsMaybe = Option(params.getArguments())
+        (argsMaybe.flatMap(_.asScala.headOption) match {
+          case Some(arg: JsonPrimitive) =>
+            val value = arg.getAsString().replace("+", " ")
+            scribe.debug(
+              s"Executing ResetChoicePopup ${params.getCommand()} for choice ${value}"
+            )
+            resetPopupChoice(value)
+          case _ =>
+            scribe.debug(
+              s"Executing ResetChoicePopup ${params.getCommand()} in interactive mode."
+            )
+            PopupChoiceReset.interactiveReset(languageClient, resetPopupChoice)
+        }).asJavaObject
+
+      case ServerCommands.ResetNotifications() =>
+        foreachSeq(_.resetNotifications(), ignoreValue = true)
+
+      case ServerCommands.NewScalaFile(args) =>
+        val directoryURI = args.lift(0).flatten
+        val name = args.lift(1).flatten
+        val fileType = args.lift(2).flatten
+        directoryURI
+          .map(getServiceFor)
+          .getOrElse(folderServices.head)
+          .handleFileCreation(directoryURI, name, fileType, isScala = true)
+      case ServerCommands.NewJavaFile(args) =>
+        val directoryURI = args.lift(0).flatten
+        val name = args.lift(1).flatten
+        val fileType = args.lift(2).flatten
+        directoryURI
+          .map(getServiceFor)
+          .getOrElse(folderServices.head)
+          .handleFileCreation(directoryURI, name, fileType, isScala = false)
+      case ServerCommands.StartAmmoniteBuildServer() =>
+        foreachSeq(_.ammoniteStart(), ignoreValue = false)
+      case ServerCommands.StopAmmoniteBuildServer() =>
+        foreachSeq(_.ammoniteStop(), ignoreValue = false)
+      case ServerCommands.StartScalaCliServer() =>
+        val f = focusedDocument match {
```

Comment:
use onCurrentFolder ?

Reply:
We use `focusedDocument` here to find the actual file not just the folder.